File: sprint1/C#/lesson5code1.cs
int[] visitors = {0, 2, 3, 2, 0, 4, 1, 1, 2};
int[] entriesByVisitor = new int[5];
int bestVisitor = 0;

foreach (int visitor in visitors)
{
    entriesByVisitor[visitor]++;
    if (entriesByVisitor[visitor] > entriesByVisitor[bestVisitor])
    {
        bestVisitor = visitor;
    }
}

Console.WriteLine(bestVisitor);

File: sprint1/JS/lesson5code1.js
// visitors - массив номеров пассажиров.
// Каждый пассажир проехал столько раз, сколько раз встречается его номер
const visitors = [0,2,3,2,0,4,1,1,2];
const entries_by_visitor = [0, 0, 0, 0, 0];
let best_visitor = 0;

// Проходим по всем пассажирам и увеличиваем счётчик их посещений
for (let i = 0; i < visitors.length; i++) {
    const visitor = visitors[i];
    entries_by_visitor[visitor] += 1;
    // Если текущий пассажир имеет больше посещений, чем лучший пассажир, обновляем лучшего пассажира
    if (entries_by_visitor[visitor] > entries_by_visitor[best_visitor]) {
        best_visitor = visitor;
    }
}

// Выводим номер лучшего пассажира
console.log(best_visitor);
